### PR TITLE
dnsdist: Add `NetmaskGroup:addNMG()` to merge Netmask groups

### PR DESCRIPTION
**Pull request by rgacogne**
### Short description

<strike>This needs a few tests.</strike>

Closes https://github.com/PowerDNS/pdns/issues/13203

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [x] included documentation (including possible behaviour changes)
- [ ] documented the code
- [x] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: pdns/dnsdist-lua-bindings.cc [hunk 1]
```diff
@@ -379,10 +379,17 @@ void setupLuaBindings(LuaContext& luaCtx, bool client)
 
   /* NetmaskGroup */
   luaCtx.writeFunction("newNMG", []() { return NetmaskGroup(); });
-  luaCtx.registerFunction<void(NetmaskGroup::*)(const std::string&mask)>("addMask", [](NetmaskGroup&nmg, const std::string& mask)
+  luaCtx.registerFunction<void(NetmaskGroup::*)(const std::string& mask)>("addMask", [](NetmaskGroup&nmg, const std::string& mask)
                          {
                            nmg.addMask(mask);
                          });
+  luaCtx.registerFunction<void(NetmaskGroup::*)(const NetmaskGroup& otherNMG)>("addMMG", [](NetmaskGroup& nmg, const NetmaskGroup& otherNMG) {
```
Comment (pieterlexis) [inline pdns/dnsdist-lua-bindings.cc:386]:
```suggestion
  luaCtx.registerFunction<void(NetmaskGroup::*)(const NetmaskGroup& otherNMG)>("addNMG", [](NetmaskGroup& nmg, const NetmaskGroup& otherNMG) {
```

Comment (rgacogne) [reply pdns/dnsdist-lua-bindings.cc]: :facepalm: Thanks! I just fixed it, and added a test in the process :)

File: pdns/dnsdist-lua-bindings.cc [hunk 2]
```diff
@@ -379,10 +380,17 @@ void setupLuaBindings(LuaContext& luaCtx, bool client)
 
   /* NetmaskGroup */
   luaCtx.writeFunction("newNMG", []() { return NetmaskGroup(); });
-  luaCtx.registerFunction<void(NetmaskGroup::*)(const std::string&mask)>("addMask", [](NetmaskGroup&nmg, const std::string& mask)
+  luaCtx.registerFunction<void(NetmaskGroup::*)(const std::string& mask)>("addMask", [](NetmaskGroup&nmg, const std::string& mask)
```
Comment (omoerbeek) [inline pdns/dnsdist-lua-bindings.cc:383]:
```suggestion
  luaCtx.registerFunction<void(NetmaskGroup::*)(const std::string& mask)>("addMask", [](NetmaskGroup& nmg, const std::string& mask)
```

File: pdns/dnsdistdist/docs/reference/netmaskgroup.rst [hunk 3]
```diff
@@ -17,6 +17,14 @@ NetmaskGroup
     :param string mask: Add this mask, prefix with `!` to exclude this mask from matching.
     :param table masks: Adds the keys of the table to the :class:`NetmaskGroup`. It should be a table whose keys are :class:`ComboAddress` objects and whose values are integers. The integer values of the table entries are ignored. The table is of the same type as the table returned by the `exceed*` functions.
 
+  .. method:: NetmaskGroup:addNMG(otherNMG)
+
+    .. versionadded:: 1.9.0
+
+    Add one or more masks from an existing to this NMG.
```
Comment (omoerbeek) [inline pdns/dnsdistdist/docs/reference/netmaskgroup.rst:24]:
```suggestion
    Add all masks from an existing NMG to this NMG.
```

**Review by omoerbeek — APPROVED** [review]
Two nits

**Comment by rgacogne** [comment]
Rebased on master to fix a conflict.

**Pull request merged**